saga/README.md

```diff
@@ -0,0 +1,121 @@
+```
+
+```console
+$ docker-compose up --build
+```
+
+Register the connectors for the different services:
+
+```console
+$ http PUT http://localhost:8083/connectors/order-outbox-connector/config < register-order-connector.json
+$ http PUT http://localhost:8083/connectors/payment-outbox-connector/config < register-payment-connector.json
+$ http PUT http://localhost:8083/connectors/credit-outbox-connector/config < register-credit-connector.json
+```
+
+Place an order:
+
+```console
+$ http POST http://localhost:8080/orders < requests/place-order.json
+```
+
+Examine the emitted messages for `payment` and `credit-approval` in Apache Kafka:
+
+```console
+$ docker run --tty --rm \
+    --network saga-network \
+    debezium/tooling:1.1 \
+    kafkacat -b kafka:9092 -C -o beginning -q \
+    -f "{\"key\":%k, \"headers\":\"%h\"}\n%s\n" \
+    -t payment.request
+
+$ docker run --tty --rm \
+    --network saga-network \
+    debezium/tooling:1.1 \
+    kafkacat -b kafka:9092 -C -o beginning -q \
+    -f "{\"key\":%k, \"headers\":\"%h\"}\n%s\n" \
+    -t credit-approval.request
+```
+
+Examine the saga state in the order service's database:
+
+```console
+$ docker run --tty --rm -i \
+        --network saga-network \
+        debezium/tooling:1.1 \
+        bash -c 'pgcli postgresql://orderuser:orderpw@order-db:5432/orderdb'
+
+select * from order.sagastate;
+
++--------------------------------------+------------------------------------------------------------------------------------------+----------+---------------------------------------------------+-----------------+-----------+
+| id                                   | payload                                                                                  | status   | stepstate                            | type            | version   |
+|--------------------------------------+------------------------------------------------------------------------------------------+----------+---------------------------------------------------+-----------------+-----------|
+| d08b4e43-8522-4539-9aba-de4bb0dc1a8e | {"payment-due":59,"customer-id":456,"order-id":2,"credit-card-no":"xxxx-yyyy-dddd-aaaa"} | COMPLETED  | {"credit-approval":"SUCCEEDED","payment":"SUCCEEDED"} | order-placement | 1         |
++--------------------------------------+------------------------------------------------------------------------------------------+----------+---------------------------------------------------+-----------------+-----------+
+```
+
+Alternatively, you also can access pgAdmin on http://localhost:5050 (pgadmin4@pgadmin.org/admin).
+
+Place an order with an invalid credit card number (the payment service rejects any number that ends with "9999"):
+
+```console
+$ http POST http://localhost:8080/orders < requests/place-invalid-order1.json
+```
+
+Observe how the saga's state is `ABORTED`, with the `payment` step in state `FAILED`, and the `credit-approval` first in step in state `ABORTING`, then `ABORTED`.
```

Comment:
at the sentence end I guess this should just read "first in state aborting, then aborted" right?

Reply:
Yes :)